espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/http/RouteMap.java

```diff
@@ -21,13 +21,12 @@ public void addRoute(RouteDefinition route) {
     }
 
     @Nullable
-    public RouteDefinition findMatchingRoute(String uri) {
-        for (Map.Entry<Method, Map<String, RouteDefinition>> methodMap: routeMap.entrySet()) {
-           for (Map.Entry<String, RouteDefinition> route: methodMap.getValue().entrySet()) {
-               RouteDefinition routeDefinition = route.getValue();
-               if (routeDefinition.isMatch(uri)) {
-                   return routeDefinition;
-               }
+    public RouteDefinition findMatchingRoute(Method method, String uri) {
+        Map<String, RouteDefinition> methodMap = routeMap.get(method);
```

Comment:
should we check if the key exists in the map first to avoid unexpected exception??